pkg/schedule/schedulers/scheduler_controller.go

```diff
@@ -470,29 +471,27 @@ func (s *ScheduleController) Schedule(diagnosable bool) []*operator.Operator {
 		if diagnosable {
 			s.diagnosticRecorder.SetResultFromPlans(ops, plans)
 		}
-		foundDisabled := false
+		if len(ops) == 0 {
+			continue
+		}
+
+		// If we have schedule, reset interval to the minimal interval.
+		s.nextInterval = s.Scheduler.GetMinInterval()
 		for _, op := range ops {
-			if labelMgr := s.cluster.GetRegionLabeler(); labelMgr != nil {
-				region := s.cluster.GetRegion(op.RegionID())
-				if region == nil {
-					continue
-				}
-				if labelMgr.ScheduleDisabled(region) {
-					denySchedulersByLabelerCounter.Inc()
-					foundDisabled = true
-					break
-				}
+			labelMgr := s.cluster.GetRegionLabeler()
+			if labelMgr == nil {
+				continue
 			}
-		}
-		if len(ops) > 0 {
-			// If we have schedule, reset interval to the minimal interval.
-			s.nextInterval = s.Scheduler.GetMinInterval()
-			// try regenerating operators
-			if foundDisabled {
+			region := s.cluster.GetRegion(op.RegionID())
+			if region == nil {
 				continue
```

Comment:
I find that if the region == nil, do we still need to count it in ops?

Reply:
https://github.com/tikv/pd/blob/c015f140f49a47e60b3884a9179e79e92482461c/pkg/schedule/labeler/labeler.go#L324

seems it will be panic

Reply:
https://github.com/tikv/pd/pull/8276/commits/267dc63cb6d33181d03a4d7c066404f4d863f1ee

`region == nil` means the ops are not normal right now, so can just retry